validator/client/validator.go

```diff
@@ -977,47 +977,67 @@ func (v *validator) UpdateFeeRecipient(ctx context.Context, km keymanager.IKeyma
 		return err
 	}
 	log.Infoln("Successfully prepared beacon proposer with fee recipient to validator index mapping.")
+	for _, request := range registerValidatorRequests {
```

Comment:
when it runs per epoch I have it running in its own go routine you can see that in the runner.go file line 190. otherwise it runs when status updates and on startup which are not on their on go routine. 

I think those are all possible future improvements, with it being set every6.4 minutes and on startup as well I wonder if that's 100% needed at this moment. I can make a note on the tracking of this feature but I don't think it should be part of this particular PR.

Reply:
> with it being set every6.4 minutes and on startup as well I wonder if that's 100% needed at this moment.
I don't follow your meaning here, what is it that may not be needed based on the frequency of this running?

Are we building a prototype for integration testing, or is this something we want to be usable by end-users once this is merged? If it's a prototype then I agree that we can punt on resiliency, but I think accounting for retries and partial failures are basic requirements we need in place before this is used in production. I'll defer to @terencechain on clarifying the goal.